flattentool/input.py

```diff
@@ -571,6 +571,16 @@ def unflatten_main_with_parser(parser, line, timezone, xml, id_name):
                 raise ValueError("There is an object or list at '{}' but it should be an {}".format(path_till_now, current_type))
 
             ## Other Types
+            current_path_value = current_path.get(path_item)
+            if not xml and (type(current_path_value) is ListAsDict or hasattr(current_path_value, 'items')):
```

Comment:
This will evaluated first `not xml and (type(current_path_value) is ListAsDict ` and then the `or` clause. That's what you want, right? Just checking.